### ISSUE TITLE
Dashboard Compare diff does not use gitignore

### ISSUE THREAD

**Issue by Mobrockers** [stryker-mutator/stryker-net#1188]
**Describe the bug**
The diff feature (on azure devops) is picking up the gitignore file and log folders stryker creates as output.

**Logs**
```
[19:30:05 INF] Changed file D:\a\1\s\src\Stryker.Core\Stryker.Core.UnitTest\StrykerOutput\.gitignore
[19:30:05 INF] Changed file D:\a\1\s\src\Stryker.Core\Stryker.Core.UnitTest\StrykerOutput\2020-09-04.19-29-19\logs\VsTest-log.host.20-09-04_19-29-59_93106_7.txt
[19:30:05 INF] Changed file D:\a\1\s\src\Stryker.Core\Stryker.Core.UnitTest\StrykerOutput\2020-09-04.19-29-19\logs\VsTest-log.txt
```

**Expected behavior**
Since they are folders and files internal to stryker and they are created for every testrun these files should not be detected as file changes by the dashboard compare feature.

**Comment by g7ed6e** [comment]
Hi, I'm facing this issue too. Any idea how to fix it ?

**Comment by Mobrockers** [comment]
Since this works fine locally and in other tools such as visual studio I think we need to fix this either in code or in libgit2sharp.

**Comment by g7ed6e** [comment]
Will à quick & dirty fix excluding stryker generated files based on a regex in the GitDiffProvider be accepted ?

**Comment by Mobrockers** [comment]
Hi @g7ed6e 

I would prefer to find a different solution. We already use a library called libgit2sharp for all git operations. I would assume it has some option to control gitignore usage. Perhaps you could look into that? If it doesn't exist in libgit2sharp then a regex based solution is acceptable.

**Comment by Mobrockers** [comment]
Workaround has been implemented in #1199 but a proper fix where libgit2sharp uses the provided gitignore file should be implemented.

**Comment by g7ed6e** [comment]
Thanks @Mobrockers for merging this quick fix. I will come back to you with a proper fix asap. I think there is no issue in libgit2sharp itself but with its usage in dotnet-stryker which is not correct in pull request scenario. I will let you know my conclusions after more tests.

**Comment by Mobrockers** [comment]
According to libgit2 devs the solution is to stop using DiffTargets.Index in the diff. The only resulting change should be that the gitignore is used now

![image](https://user-images.githubusercontent.com/3595932/94950321-c0480f00-04e2-11eb-9759-8edb46a050d7.png)

**Comment by g7ed6e** [comment]
@Mobrockers great ! Considering local changes like generated files did not make sense to me. I propose to handle this fix in the temp branch you created to fix the GitInfoProvider as there bugs are closely tied. Do you agree?

**Comment by Mobrockers** [comment]
Well I was going to close my temp branch and just push the fixes to yours. But I would rather fix this separately because that makes it easier to see the effect of the change.

**Comment by g7ed6e** [comment]
@Mobrockers i implemented the suggested changes there #1235
